mhkit/tidal/io/noaa.py

```diff
@@ -84,7 +88,82 @@ def request_noaa_data(
         Request metadata. If returning xarray, metadata is instead attached to
         the data's attributes.
     """
```

Comment:
git compares the changes in `request_noaa_data` poorly, but I'm more or less following how things are broken up into multiple functions